File: QuanLyPhongKham.Server/src/QuanLyPhongKham.DTOs/Requests/NhanVienRequest.cs
using QuanLyPhongKham.Commons.Enum;
using System.ComponentModel.DataAnnotations;

namespace QuanLyPhongKham.DTOs.Requests
{
    public class NhanVienRequest
    {
        public Guid MaNV { get; set; }

        [Required(ErrorMessage = "Họ tên là bắt buộc.")]
        [MaxLength(50, ErrorMessage = "Họ tên không vượt quá 100 ký tự..")]
        public string HoTen { get; set; }

        [Required(ErrorMessage = "Email là bắt buộc.")]
        [EmailAddress(ErrorMessage = "Email khong hop le.")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Số điện thoại là bắt buộc.")]
        [MaxLength(10, ErrorMessage = "Số điện thoại không vượt quá 10 ký tự.")]
        [Phone(ErrorMessage = "Số điện thoại không đúng định dạng.")]
        public string? SoDienThoai { get; set; }

        [Required(ErrorMessage = "Mật khẩu là bắt buộc.")]
        [StringLength(20, MinimumLength = 6, ErrorMessage = "Mật khẩu có ít nhất là 6 ký tự.")]
        [DataType(DataType.Password)]
        public string Password { get; set; }

        [Required(ErrorMessage = "Vai trò là bắt buộc.")]
        public string Role { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/interfaces/requests/NhanVienRequest.js
/**
 * @typedef {Object} NhanVienRequest
 * @property {string} maNV - Employee ID (for updates)
 * @property {string} hoTen - Full name
 * @property {string} email - Email
 * @property {string} soDienThoai - Phone number
 * @property {string} password - Password
 * @property {string} role - Role
 */

export const createNhanVienRequest = (
	maNV = '',
	hoTen = '',
	email = '',
	soDienThoai = '',
	password = '',
	role = ''
) => ({
	maNV,
	hoTen,
	email,
	soDienThoai,
	password,
	role,
});